File: 2819_minimum_relative_loss_after_buying_chocolates/solution.py
# LeetCode 2819 - Minimum Relative Loss After Buying Chocolates
# https://leetcode.com/problems/minimum-relative-loss-after-buying-chocolates/

from typing import List


class Solution:
    def minimumRelativeLosses(self, prices: List[int], queries: List[List[int]]) -> List[int]:
        prices = sorted(prices)
        n = len(prices)
        ans = [0] * len(queries)
        for qi, (kk, m) in enumerate(queries):
            losses = [0] * n
            for i in range(n):
                if prices[i] <= kk:
                    losses[i] = prices[i]
                else:
                    losses[i] = 2 * kk - prices[i]
            losses.sort()
            total = 0
            for i in range(m):
                total += losses[i]
            ans[qi] = total
        return ans

File: 2819_minimum_relative_loss_after_buying_chocolates/Solution.swift
// LeetCode 2819 - Minimum Relative Loss After Buying Chocolates
// https://leetcode.com/problems/minimum-relative-loss-after-buying-chocolates/

class Solution {
    func minimumRelativeLosses(_ prices: [Int], _ queries: [[Int]]) -> [Int] {
        let prices = prices.sorted()
        let n = prices.count
        return queries.map { q in
            let kk = q[0], m = q[1]
            var losses = prices.map { $0 <= kk ? $0 : 2 * kk - $0 }
            losses.sort()
            return losses.prefix(m).reduce(0, +)
        }
    }
}
